packages/compiler/src/expression_parser/parser.ts

```diff
@@ -340,10 +340,30 @@ export class Parser {
 
     return errLocation.length;
   }
+
+  /** Like `String.prototype.indexOf`, but skips content inside quotes. */
+  private _indexOfSkipQuoted(input: string, value: string, start: number): number {
+    const valueLength = value.length;
+    let currentQuote: string|null = null;
+    for (let i = start; i < input.length; i++) {
+      const char = input[i];
+      // Skip the characters inside quotes. Note that we only care about the
+      // outer-most  quotes matching up and we need to account for escape characters.
+      if (isQuote(input.charCodeAt(i)) && (currentQuote === null || currentQuote === char) &&
+          input[i - 1] !== '\\') {
+        currentQuote = currentQuote === null ? char : null;
+      } else if (
+          currentQuote === null && char === value[0] &&
+          (valueLength === 0 || input.substring(i, i + valueLength) === value)) {
```

Comment:
I don't think `valueLength` can be zero here, since that would mean that `char === undefined` in `char === value[0]`. Also I think you can use `substr()` simplify.

```suggestion
          input.substr(i, valueLength) === value)) {
```

Reply:
Or just `input.startsWith(value, i)` ;)

Reply:
Ooh! I didn't know that `startsWith()` can have a `position` argument. I had considered suggesting `indexOf()` with a position but obviously this could end up searching long strings over and over again so is not viable.

Great call @JoostK